Core/Business/Addon/AddonInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * 2007-2015 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Business\Addon;
+
+interface AddonInterface
+{
+    public function enable();
+    public function disable();
+
+    public function upgrade(string $version);
+    public function rollback(string $version);
```

Comment:
We can use the existing modules as an example. Sometimes, a module has an upgrade file to execute when it is updated. If we want to rollback them, we won't able to execute the same files in a different order (the sames changes would be applied two times).

This discussion makes me think about a potential idea for this action. If a module as an upgrade file, it should implement a new function in order to be rollbacked succesfully.

### Example file upgrade/install-1.1.0.php
```php
function upgrade_module_1_1_0($object)
{
	return Db::getInstance()->execute(/* SQL request altering the database */);
}

function rollback_module_1_1_0($object)
{
	return Db::getInstance()->execute(/* SQL request reverting the changes */);
}
```